packages/legacy-scripting-runner/middleware-factory.js

```diff
@@ -20,35 +20,51 @@ const renderTemplate = (templateString, context) => {
   return _.template(templateString, options)(finalContext);
 };
 
+const getLowerHeaders = (headers) =>
+  Object.entries(headers).reduce((result, [k, v]) => {
+    result[k.toLowerCase()] = v;
+    return result;
+  }, {});
+
+const applyAuthMappingInHeaders = (authMapping, req, authData) => {
+  const lowerHeaders = getLowerHeaders(req.headers);
+  Object.entries(authMapping).forEach(([k, v]) => {
+    const lowerKey = k.toLowerCase();
+    if (!lowerHeaders[lowerKey]) {
+      req.headers[k] = renderTemplate(v, authData);
```

Comment:
Seems like we should always use a header/value from `authData`, and discard/replace an existing header that's already present within the request?  

```js
req.headers.secret_key = 'some default value';
usersAuthData = { secret_key: "the user's stored/saved auth key" };
applyAuthMappingInHeaders(mapping, req, usersAuthData);
```

Reply:
Great callout! I did another test on WB. WB seems to always override the request headers with `Content-Type: application/json; charset=utf-8` and `Accept: application/json`. So regardless what `Content-Type` the `authData` may have, the header is always `Content-Type: application/json; charset=utf-8` before the bundle goes into a `pre_(poll|write|search)` method.